lib/hexpm/web/controllers/api/package_controller.ex

```diff
@@ -3,8 +3,14 @@ defmodule Hexpm.Web.API.PackageController do
 
   plug :fetch_repository when action in [:index]
   plug :maybe_fetch_package when action in [:show]
-  plug :maybe_authorize, [domain: "api", fun: &maybe_repository_access/2] when action in [:index]
-  plug :maybe_authorize, [domain: "api", fun: &repository_access/2] when action in [:show]
+
+  plug :maybe_authorize,
+       [domain: "api", resource: "read", fun: &maybe_repository_access/2]
+       when action in [:index]
+
+  plug :maybe_authorize,
+       [domain: "api", resource: "read", fun: &repository_access/2]
```

Comment:
The authorize/access system isn't great, it definitely needs a refactor or some kind of improvement, it was built before we had private repos.

We use `repository_access` on `show` because we wan't to return 403 for unknown repositories names instead of 404. So if a private repository is accessed it's a hard auth error.

We use `maybe_repository_access` on `index` because it should not necessarily be an auth error if you have not authenticated, i.e. no API key was given. The tricky part with this endpoint is that you have both `api/packages` and `api/repos/:repository/packages` and the former route changes behaviour if you are authed or not.